subsys/bluetooth/host/scan.c

```diff
@@ -27,6 +28,41 @@ static bt_le_scan_cb_t *scan_dev_found_cb;
 static sys_slist_t scan_cbs = SYS_SLIST_STATIC_INIT(&scan_cbs);
 
 #if defined(CONFIG_BT_EXT_ADV)
+/* A buffer pool used to reassemble advertisement data from the controller. */
+NET_BUF_POOL_FIXED_DEFINE(ext_scan_buf_pool, 1, CONFIG_BT_EXT_SCAN_BUF_SIZE, 8, NULL);
+static struct net_buf *ext_scan_buf;
+
+struct fragmented_advertiser {
+	bt_addr_le_t addr;
+	uint8_t sid;
+	struct net_buf *reassembly_buffer;
+};
+
+static struct fragmented_advertiser reassembling_advertiser;
+static struct fragmented_advertiser *p_reassembling_advertiser;
```

Comment:
I am mostly using it as a flag to see if we are currently reassembling anything. A bool flag in the struct may be better.

Reply:
I am thinking if we add an enum for state to the `fragmented_advertiser` struct we can get rid of this pointer and the `reassembly_buffer` pointer. Something like:
```c
enum {
  INACTIVE,
  REASSEMBLING,
  DISCARDING,
} state;
```

Currently `p_reassembling_advertiser` is set to `NULL` to signal that the advertiser is inactive and to `&reassembling_advertiser` to signal that it is reassembling and the `reassembly_buffer` pointer is set to `NULL`  to signal that the advertiser is still active, but the data should be discarded.

Reply:
@hermabe Could we use the buf->len == 0 to determine if a reassembly is in progress?

Reply:
That should work. Still need to flag if the reports should be discarded.

Reply:
Yes :) The change that I suggest here is that instead of checking `p_reassembling_advertiser == NULL` we check `reassembling_advertiser.reassembly_buffer->len == 0` (possibly as a macro).

Reply:
When `reassembling_advertiser.reassembly_buffer` is set to `NULL` to signal that reports should be discarded this does not work. Created the enum mentioned above instead.